### ISSUE TITLE
`flutter test` should allow me to use the json reporter available with `pub run test --reporter json`

### ISSUE THREAD

**Issue by dustin-graham** [flutter/flutter#26769]
I am making improvements on build automation for our app and one of my tasks is to generate reports when our automated tests run and send them to our test management software. To do this I need to be able to generate the json reports that are available when running tests via `pub run test --reporter json`. This is noted here: https://github.com/dart-lang/test/blob/master/pkgs/test/doc/json_reporter.md.

However, the `flutter test` tool does not give this option. I attempted to run the tests via `flutter packages pub run test --reporter json` but this does not work with the flutter tests. As it stands right now, my pipeline can tell whether or not the tests have all passed or not but I can't provide any additional insights beyond that at this time.

I did spend a good amount of time searching for an answer to my question, and I think https://github.com/flutter/flutter/issues/18455 may be related but I'm not exactly sure what the context is for that issue.

In summary, I'm looking for a way to generate a machine-readable test report when running my flutter tests.

I appreciate your time and thank you all very much for giving us this incredible tool.

**Comment by jonahwilliams** [comment]
Does `flutter test --machine` meet your requirements? It returns a structured JSON output used by IDEs.

**Comment by dustin-graham** [comment]
Oh, thank you. Didn't know about that one. I'll give that a try. Thank you!

**Comment by dustin-graham** [comment]
@jonahwilliams thanks for the tip. that was exactly what I was looking for. Thank you.
